tools/monitoring/landsat_gap_report.py

```diff
@@ -1 +1,299 @@
+
+        if len(missing_scenes) > 0:
+            output_filename = (
+                f"{satellite_name}_{date_string}.txt.gz"
+                if not update_stac
+                else f"{satellite_name}_{date_string}_update.txt.gz"
+            )
+
+            log.info(
+                f"File will be saved in {URL(landsat_status_report_path) / output_filename}"
+            )
+            s3_dump(
+                data=gzip.compress(str.encode("\n".join(missing_scenes))),
+                url=str(URL(landsat_status_report_path) / output_filename),
+                s3=landsat_s3,
+                ContentType="application/gzip",
+            )
+
+            log.info(f"Number of missing scenes: {len(missing_scenes)}")
+
+        if len(orphaned_scenes) > 0:
+            orphan_output_filename = f"{satellite_name}_{date_string}_orphaned.txt.gz"
+            s3_dump(
+                data=gzip.compress(str.encode("\n".join(orphaned_scenes))),
+                url=str(URL(landsat_status_report_path) / orphan_output_filename),
+                s3=landsat_s3,
+                ContentType="application/gzip",
+            )
+
+            log.info(f"Number of orphaned scenes: {len(orphaned_scenes)}")
+
+        message = dedent(
+            f"*{satellite_name.upper()} GAP REPORT*\n "
+            f"Environment: {environment}\n "
+            f"Missing Scenes: {len(missing_scenes)}\n"
+            f"Orphan Scenes: {len(orphaned_scenes)}\n"
+            f"Missing Scenes report Saved: {str(URL(landsat_status_report_path) / output_filename)}\n"
+            f"Orphan Scenes report Saved: {str(URL(landsat_status_report_path) / orphan_output_filename)}\n"
+        )
+
+        log.info(message)
+
+        if len(missing_scenes) > 0 or len(orphaned_scenes) > 0:
+
+            log.info(
+                f"File {file_name} processed and sent in {time_process(start=start_timer)}"
+            )
+
+            if notification_url is not None:
+                send_slack_notification(
+                    notification_url, f"{satellite_name} Gap Report", message
+                )
+
+            if not update_stac and (
+                len(missing_scenes) > 200 or len(orphaned_scenes) > 200
+            ):
+                raise Exception(f"More than 200 scenes were found \n {message}")
+
+    except Exception as error:
+        log.exception(error)
+        raise error
+
+
+@click.argument(
+    "bucket_name",
+    type=str,
+    nargs=1,
+    required=True,
+    default="Bucket where the gap report is",
+)
+@click.argument(
+    "satellite",
+    type=str,
+    nargs=1,
+    required=True,
+    default="satellite to be compared, supported ones (landsat_8, landsat_7, landsat_5)",
+)
+@update_stac
+@slack_url
+@click.command("landsat-gap-report")
+def cli(
+    bucket_name: str, satellite: str, update_stac: bool = False, slack_url: str = None
+):
+    """
+    Publish missing scenes
+    """
+    generate_buckets_diff(
```

Comment:
Perhaps this shouldn't do the notification, and the function should return diffs, like it says it does.

Then this `cli` function can do the notification and sys.exit(1) for when there's too many.